umap/static/umap/js/umap.forms.js

```diff
@@ -804,18 +804,67 @@ L.FormBuilder.FacetSearch = L.FormBuilder.Element.extend({
       label = L.DomUtil.create('label', '', property_li)
     input.type = 'checkbox'
     input.id = `checkbox_${this.name}_${value}`
-    input.checked = this.get().includes(value)
+    input.checked = this.get()['choices'].includes(value)
     input.dataset.value = value
     label.htmlFor = `checkbox_${this.name}_${value}`
     label.innerHTML = value
     L.DomEvent.on(input, 'change', (e) => this.sync())
   },
 
   toJS: function () {
-    return [...this.ul.querySelectorAll('input:checked')].map((i) => i.dataset.value)
+    return {
+      'type': 'checkbox',
+      'choices': [...this.ul.querySelectorAll('input:checked')].map((i) => i.dataset.value)
+    }
   },
 })
 
+L.FormBuilder.FacetSearchDate = L.FormBuilder.Element.extend({
+  build: function () {
+    this.container = L.DomUtil.create('div', 'umap-facet', this.parentNode);
+    const min = this.options.criteria['min'];
+    const max = this.options.criteria['max'];
+
+    // Create labels for min and max inputs
+    this.minLabel = L.DomUtil.create('label', '', this.container);
+    this.minLabel.innerHTML = 'Start';
```

Comment:
Maybe, instead of `Start`/`End` we could have `From` and `to`  in the same line ?

Reply:
Start and End were supposed to be generic and also usable for numbers but since we have to have an own L.FormBuilder.Element deviate for those, we could also have nicer labels. I would go with From and Until for dates and times and Min and Max for numbers.

I also enhanced the implementation of L.FormBuilder.FacetSearchDate with a table. Looks nicer now.

![image](https://github.com/umap-project/umap/assets/56305172/d67c47f6-2fec-4d19-a666-5c190e354039)